src/vercel/services/vercel.service.ts

```diff
@@ -23,9 +35,133 @@ export class VercelService {
           name: domainName,
         },
       })
-    } catch (error: any) {
-      this.logger.error('Error adding domain:', error)
-      throw error
+    } catch (error) {
+      this.logger.error(`Error adding domain ${domainName} to project:`, error)
+      throw new Error(`Failed to add domain to Vercel project: ${error}`)
+    }
+  }
+
+  async removeDomainFromProject(domainName: string) {
+    try {
+      return await this.client.projects.removeProjectDomain({
+        idOrName: VERCEL_PROJECT_ID!,
+        domain: domainName,
+        teamId: VERCEL_TEAM_ID,
+      })
+    } catch (error) {
+      this.logger.error(
+        `Error removing domain ${domainName} from project:`,
+        error,
+      )
+      throw new Error(`Failed to remove domain from Vercel project: ${error}`)
+    }
+  }
+
+  async verifyProjectDomain(domainName: string) {
+    try {
+      return await this.client.projects.verifyProjectDomain({
+        idOrName: VERCEL_PROJECT_ID!,
+        domain: domainName,
+        teamId: VERCEL_TEAM_ID,
+      })
+    } catch (error) {
+      this.logger.error(`Error verifying domain ${domainName}:`, error)
+      throw new Error(`Failed to verify domain: ${error}`)
+    }
+  }
+
+  async checkDomainPrice(domainName: string) {
+    try {
+      const result = await this.client.domains.checkDomainPrice({
+        name: domainName,
+        teamId: VERCEL_TEAM_ID,
+      })
+
+      this.logger.debug(`Price check for ${domainName}:`, result)
+      return result
+    } catch (error) {
+      this.logger.error(`Error checking price for domain ${domainName}:`, error)
+      throw new Error(`Failed to check domain price: ${error}`)
+    }
+  }
+
+  /**
+   * Purchase a domain through Vercel
+   * @param domainName - The domain name to purchase (e.g. 'example.com')
+   * @param contact - Contact information for domain registration
+   * @param expectedPrice - The expected price for the domain
+   * @returns Operation result from Vercel
+   */
+  async purchaseDomain(
+    domainName: string,
+    contact: {
+      firstName: string
+      lastName: string
+      email: string
+      phoneNumber: string
+      addressLine1: string
+      addressLine2?: string
+      city: string
+      state: string
+      zipCode: string
+    },
+    expectedPrice: number,
+  ) {
+    try {
+      this.logger.debug(`Purchasing domain ${domainName} through Vercel`)
+
+      const finalContact = {
+        firstName: contact.firstName || GP_DOMAIN_CONTACT.firstName,
+        lastName: contact.lastName || GP_DOMAIN_CONTACT.lastName,
+        email: contact.email || GP_DOMAIN_CONTACT.email,
+        phone: contact.phoneNumber || GP_DOMAIN_CONTACT.phoneNumber,
+        address1: contact.addressLine1 || GP_DOMAIN_CONTACT.addressLine1,
+        city: contact.city || GP_DOMAIN_CONTACT.city,
+        state: contact.state || GP_DOMAIN_CONTACT.state,
+        postalCode: contact.zipCode || GP_DOMAIN_CONTACT.zipCode,
+      }
```

Comment:
This seems duplicative.. Shouldn't this be the same logic as what's in `DomainPurchaseHandler::buildContactInfo`? 🤔 

Prolly apropos to my comment there.